ingestors/kafka/src/main/scala/hydra/kafka/programs/CreateTopicProgram.scala

```diff
@@ -232,35 +233,82 @@ final class CreateTopicProgram[F[_]: Bracket[*[_], Throwable]: Sleep: Logger](
     } else None
   }
 
+  private def nonEventStreamTypeValidation(schemas: Schemas, subject: Subject): F[Unit] =
+    (schemas.key.getType, schemas.value.getType) match {
+      case (Schema.Type.RECORD, Schema.Type.RECORD) =>
+        val keyFields = schemas.key.getFields.asScala.toList
+        val valueFields = schemas.value.getFields.asScala.toList
+        val keyFieldIsEmpty: Option[IncompatibleSchemaException] = if (keyFields.isEmpty) {
+          IncompatibleSchemaException("Must include Fields in Key").some
+        } else None
+        val validationErrorsF: F[List[RuntimeException]] = for {
+          k <- validateKeySchemaEvolution(schemas, subject)
+          v <- validateValueSchemaEvolution(schemas, subject)
+        } yield {
+          val keyFieldsCheckedForNull = checkForNullableKeyFields(keyFields)
+          val valueNullableFieldCheckedForDefault = checkForDefaultNullableValueFields(valueFields)
+          List[Option[RuntimeException]](keyFieldIsEmpty,
+            checkForMismatches(keyFields, valueFields),
+            keyFieldsCheckedForNull,
+            valueNullableFieldCheckedForDefault,
+            k,
+            v).flatten
+        }
+        validationErrorsF.flatMap { validationErrors =>
+          if (validationErrors.nonEmpty) Bracket[F, Throwable].raiseError(ValidationErrors(validationErrors)) else Bracket[F, Throwable].pure(())
+        }
+      case _ => Bracket[F, Throwable].raiseError(KeyAndValueNotRecordType)
+    }
+
+  private def eventStreamTypeValidation(schemas: Schemas, subject: Subject): F[Unit] = {
+    val validationErrors = (schemas.key.getType, schemas.value.getType) match {
+      case (Schema.Type.RECORD, Schema.Type.RECORD) =>
+        val keyFields = schemas.key.getFields.asScala.toList
+        val valueFields = schemas.value.getFields.asScala.toList
+        val keyFieldIsEmpty: Option[IncompatibleSchemaException] = if (keyFields.isEmpty) {
+          IncompatibleSchemaException("Must include Fields in Key").some
+        } else None
+        for {
+          k <- validateKeySchemaEvolution(schemas, subject)
+          v <- validateValueSchemaEvolution(schemas, subject)
+        } yield {
+          val valueNullableFieldCheckedForDefault = checkForDefaultNullableValueFields(valueFields)
+          List[Option[RuntimeException]](valueNullableFieldCheckedForDefault,
+            keyFieldIsEmpty,
+            checkForMismatches(keyFields, valueFields),
+            k,
+            v).flatten
+        }
+      case (Schema.Type.STRING, Schema.Type.RECORD) =>
+        val keyInfo = SchemaBuilder
+          .record("fakeName")
+          .fields()
+          .name(schemas.key.getName)
+          .`type`()
+          .stringType()
+          .noDefault()
+          .endRecord().getFields.asScala.toList
```

Comment:
There's not a super easy way to make a Field obj, so this was what I came up with. Lmk if there is a better way